src/main/java/org/jitsi/jicofo/AbstractChannelAllocator.java

```diff
@@ -145,9 +151,15 @@ public void run()
         catch (Throwable e)
         {
             logger.error("Channel allocator failed: ", e);
+            cancel();
         }
         finally
         {
+            if (canceled && colibriChannels != null)
+            {
+                bridgeSession.colibriConference.expireChannels(colibriChannels);
```

Comment:
There's a path that expires the channels:
https://github.com/jitsi/jicofo/blob/master/src/main/java/org/jitsi/jicofo/ParticipantChannelAllocator.java#L204

I don't think `colibriConference.expireChannels` is idempotent (I'm pretty sure it will at least send another colibri request). Have you tested this case?

Reply:
I haven't tested this case but the bridge handles it gracefully [here](https://github.com/jitsi/jitsi-videobridge/blob/06b3dfe5d53306b6959c7bfb65a53d7fc4c0bf56/src/main/java/org/jitsi/videobridge/shim/ContentShim.java#L291), as it should.